pkg/detectors/agora/agora_test.go

```diff
@@ -11,12 +11,92 @@ import (
 
 	"github.com/google/go-cmp/cmp"
 	"github.com/google/go-cmp/cmp/cmpopts"
-	"github.com/trufflesecurity/trufflehog/v3/pkg/detectors"
 
 	"github.com/trufflesecurity/trufflehog/v3/pkg/common"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/detectors"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/engine/ahocorasick"
 	"github.com/trufflesecurity/trufflehog/v3/pkg/pb/detectorspb"
 )
 
+func TestAgora_Pattern(t *testing.T) {
```

Comment:
**Question:** Agora detector key and secret have same pattern and because of that it actually picks same string twice. Is it expected or we need to add some logic to handle this. You can see in test cases that the want is actually a duplicate string.

Reply:
@kashifkhan0771 great catch. Because having a `secret` _and_ `key` as the same string _should_ be impossible, we should include some logic to detect if `secret==key` and skip that case.